build.js

```diff
@@ -5,16 +5,12 @@ var rimraf = require('rimraf');
 var acorn = require('acorn');
 var walk = require('acorn/dist/walk');
 
-var ids = [];
+var idx = process.env.SOURCE_DATE_EPOCH || parseInt(Math.random() * 1000, 10);
```

Comment:
Unless we set `SOURCE_DATE_EPOCH`, which we're not going to do, you're not going to be able to reproduce our build, which is what I assume you're going to want to do.  The `parseInt` belongs around `SOURCE_DATE_EPOCH` which is the bit that's actually a string.  You can just use `Math.floor` or `Math.round` to convert the floating point number returned by `Math.random` into an integer.

My advice would be to have this read in all the files in src and use them to generate a sha512 hash, then base64 encode that and remove characters that don't match `[a-zA-Z0-9]`.  This should give you a number of fairly random seeming (but consistent as long as the src doesn't change) characters. You can then have the `getIdFor` function take the first unused character from that string, and failing that the first un-used pair from that string, and failing that the first un-used tripple from that string and so on. That should comfortably provide enough identifiers given that we only really need about 5.

Reply:
> Unless we set SOURCE_DATE_EPOCH, which we're not going to do, you're not going to be able to reproduce our build

Not sure why you would want to reproduce *your* build per-se? Thse use-case is people wanting to reproduce a specific build given a known build environment (ie. the value of `SOURCE_DATE_EPOCH`).

> My advice would be to have this read in all the files in src and use them to generate a sha512 hash, then base64 encode [..]

Whilst this would certianly work, this seems a somewhat excessive amount of work for such a smallish sub-feature.. :)

Reply:
> Not sure why you would want to reproduce your build 

We already publish a build, so there’s no reason to need a way to build it yourself, other than verifying that out build is genuinely the result of building from source. People might want to do that to check we didn’t introduce a backdoor into the compiled code.  To do that, they would need to reproduce **our** build. I can’t think why anybody would need to reproduce a separate build.

Reply:
> I can’t think why anybody would need to reproduce a separate build.

In Debian, or example. Or anyone rebuilding from source, really... :)

Reply:
Why would anyone build from source? This isn’t platform dependent, everyone is going to want to run JavaScript at the end of this.

Reply:
> Why would anyone build from source?

Hm? Why would they not? :p

Reply:
They wouldn't because we already provide pre-built versions. There is no reason to build from source. It is up to you to provide reasons why you want to build this module from source, not me to tell you why I think you don't need it.  This is your feature request, it's up to you to justify its value!  Without some proper justification, I'm going to lose interest in this discussion.

Reply:
In general, building from source is the part that empowers all the free/open source software world to do the things they do. E.g. in openSUSE we try to build everything from source and sometimes we even add patches on top of upstream sources.